lib/api/funnel.js

```diff
@@ -244,13 +260,25 @@ class Funnel {
     if (!this.kuzzle.router.isConnectionAlive(request.context)) {
       debug('Client connection dead: dropping request: %a', request.input);
       callback(processError.get('connection_dropped'), request);
-      return 0;
+      return 1;
     }
 
     debug('Starting request %s: %a', request.id, request.input);
 
+    let _request;
+
     this.checkRights(request)
```

Comment:
Why you didn't use `async/await` instead of a callback chain here?

Reply:
Because the `execute` method is still behind the concurrent request limit wall, it mustn't create a promise until a slot is available.

This promise chain is created only after we request for a slot.

And we don't return it either because this method's callers don't expect a promise.